cpp/src/groupby/sort/group_nth_element.cu

```diff
@@ -0,0 +1,190 @@
+          thrust::make_tuple(group_offsets.begin(), group_offsets.begin() + 1)),
+        [n] __device__(auto g) ->thrust::pair<size_type, bool> {
+          auto group_offset = thrust::get<0>(g);
+          auto group_size = thrust::get<1>(g) - group_offset;
+          bool nth_within_group = (n < 0) ? group_size >= (-n) : group_size > n;
+          return thrust::pair<size_type, bool>{
+            group_offset + ((n < 0) ? group_size + n : n), nth_within_group};
+        });
+    auto nth_offset = thrust::make_transform_iterator(
+        nth_offset_and_bounds,
+        [] __device__(auto of) { return of.first; });
+    
+    auto values_view = column_device_view::create(values, stream);
+    auto values_element = thrust::make_transform_iterator(
+      thrust::counting_iterator<size_type>(0),
+      [col=*values_view] __device__(auto i) { return col.element<T>(i);});
+    // copy nth if within group bounds to result
+    thrust::gather_if(rmm::exec_policy(stream)->on(stream), 
+                      nth_offset, nth_offset + num_groups,
+                      nth_offset_and_bounds,
+                      values_element,
+                      result->mutable_view().data<ResultType>(),
+                      []__device__(auto ofs) { return ofs.second; });
+    // result nullmask = n within bounds and values.valid
+    rmm::device_buffer result_bitmask;
+    size_type result_null_count;
+    std::tie(result_bitmask, result_null_count) =
+        experimental::detail::valid_if(
+            nth_offset_and_bounds, nth_offset_and_bounds + num_groups,
+            [v=*values_view] __device__ (auto ofs) {
+              auto nth_offset = ofs.first;
+              auto nth_within_group = ofs.second;
+              //return null if nth outside of group or value is null.
+              return nth_within_group && v.is_valid(nth_offset);
+            }, stream, mr);
+    if(result_null_count)
+      result->set_null_mask(std::move(result_bitmask), result_null_count);
+    return result;
+  }
+  //skip nulls (equivalent to pandas nth(dropna='any'))
+   template <typename T>
+  std::enable_if_t<is_fixed_width<T>() && !include_nulls, std::unique_ptr<column>>
+  operator()(column_view const &values,
+             rmm::device_vector<size_type> const& group_labels,
+             rmm::device_vector<size_type> const &group_offsets,
+             size_type num_groups, size_type n,
+             rmm::mr::device_memory_resource *mr, cudaStream_t stream)
+  {
+    using ResultType = cudf::experimental::detail::target_type_t<
+        T, experimental::aggregation::Kind::NTH_ELEMENT>;
+    std::unique_ptr<column> result = make_fixed_width_column(
+                                data_type(type_to_id<ResultType>()), num_groups,
+                                mask_state::UNALLOCATED, stream, mr);
+    if (num_groups == 0) {
+      return result;
+    }
+    auto group_sizes = group_count(values, group_labels, num_groups, mr, stream);
```

Comment:
Prefer not to call it from here and use the cached results. Count is highly likely to be already present in cache as it is widely required by other aggs. If calling this depends on `include_nulls` parameter then it is ok to put that logic in `store_result_functor`.

Reply:
As mentioned in the PR description, this is a workaround until PR https://github.com/rapidsai/cudf/pull/4159 is done.

Reply:
replaced with count agg. but still a hack in place until PR #4159 is done.